pkg/skaffold/build/jib/jib.go

```diff
@@ -86,10 +86,16 @@ type filesLists struct {
 }
 
 // watchedFiles maps from project name to watched files
-var watchedFiles = map[string]filesLists{}
+var watchedFiles = map[projectKey]filesLists{}
 
-func GetBuildDefinitions(a *latest.JibArtifact) []string {
-	return watchedFiles[a.Project].BuildDefinitions
+type projectKey string
+
+func getProjectKey(workspace string, a *latest.JibArtifact) projectKey {
+	return projectKey(workspace + "+" + a.Project)
```

Comment:
As discussed, the fix in this PR make sense. I'll just let other Go folks to approve this particular implementation.